### PR TITLE
Create README to better describe project purpose

### LINKED ISSUE #35: Update README
- purpose of this repo
- ownership / maintenance
- feature readiness status
- link to dev/user instructions

### PR DESCRIPTION
**Pull request by ash211**
Fixes #35

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+
+This is a collaborative effort by several folks from different companies who are interested in seeing this feature be successful.  Companies active in this project include (alphabetically):
+
+- Google
+- Haiwen
+- Hyperpilot
+- Intel
+- Palantir
+- Pepperdata
+- Red Hat
```
Comment (ash211) [inline README.md:33]: if I've left anyone off, or anyone wants this list changed, please let me know either here or in private

File: README.md [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+
+## Getting Started
+
+- [Usage guide](docs/running-on-kubernetes.md) shows how to run the code
+- [Development docs](resource-managers/kubernetes/README.md) shows how to get set up for development
+- Code is primarily located in the [resource-managers/kubernetes](resource-managers/kubernetes) module
+
+## Why does this fork exist?
+
+Adding native integration for a new cluster manager is a large undertaking.  If poorly executed, it could introduce bugs into Spark when run on other cluster managers, cause release blockers slowing down the overall Spark project, or require hotfixes which divert attention away from development towards managing additional releases.  Any work this core in Spark needs to be done carefully to minimize the risk of those negative externalities.
```
Comment (mccheah) [inline README.md]: "Any work this core in Spark" - what does this mean?

**Review by mccheah — CHANGES_REQUESTED** [review]
Small wording thing but overall seems ok.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,41 @@
...
+
+## What is this?
+
+This is a collaboratively maintained project working on [SPARK-18278](https://issues.apache.org/jira/browse/SPARK-18278). The goal is to bring native support for Spark to use Kubernetes as a cluster manager, in a fully supported way on par with the Spark Standalone, Mesos, and Apache YARN cluster managers.
+
+## Getting Started
+
+- [Usage guide](docs/running-on-kubernetes.md) shows how to run the code
+- [Development docs](resource-managers/kubernetes/README.md) shows how to get set up for development
+- Code is primarily located in the [resource-managers/kubernetes](resource-managers/kubernetes) module
```
Comment (foxish) [inline README.md]: nit: We have multiple modules within that directory, but I think you don't mean module here in the maven sense.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,41 @@
...
+
+## Why does this fork exist?
+
+Adding native integration for a new cluster manager is a large undertaking.  If poorly executed, it could introduce bugs into Spark when run on other cluster managers, cause release blockers slowing down the overall Spark project, or require hotfixes which divert attention away from development towards managing additional releases.  Any work this core in Spark needs to be done carefully to minimize the risk of those negative externalities.
+
+At the same time, an increasing number of people from varying companies and organizations desire to work together to natively run Spark on Kubernetes.  The group needs a code repository, communication forum, issue tracking, and continuous integration, all in order to work together effectively on an open source product.
```
Comment (foxish) [inline README.md]: s/varying/various/?

File: README.md (hunk 3 above)
Comment (ash211) [reply README.md]: using folder instead

File: README.md (hunk 2 above)
Comment (ash211) [reply README.md]: Changed to "Any work this deep inside Spark"

File: README.md (hunk 4 above)
Comment (ash211) [reply README.md]: done

**Comment by ash211** [comment]
Responded to both your comments @foxish and @mccheah .  Anything else?

**Comment by foxish** [comment]
LGTM

**Comment by mccheah** [comment]
Looks great! I'm going to merge.

**Review by mccheah — APPROVED** [review]

**Pull request merged**